### ISSUE TITLE
ManagerCron registerTask does not correctly keep track of run event handler

### ISSUE THREAD

**Issue by JSti** [athombv/apps-sdk-issues#50]
It seems the registerTask method of ManagerCron returns a new CronTask instance with a run event handler reference shared among all registered tasks instead of keeping track of one for each registered task. I don't think this is intended.

Steps to illustrate/reproduce problem:

```
let date = new Date();

date.setSeconds(date.getSeconds() + 3);

ManagerCron.registerTask('task1', date, { payload: 'task1 payload' })
    .then((task) => {
	    task.on('run', (data) => {
	    	console.log('Output from task1', data);
	    });
    });

date.setSeconds(date.getSeconds() + 3);

ManagerCron.registerTask('task2', date, { payload: 'task2 payload' })
	.then((task) => {
	    task.on('run', (data) => {
	    	console.log('Output from task2', data);
	    });
	});
```

Expected output:
```
Output from task1 { payload: 'task1 payload' }
Output from task2 { payload: 'task2 payload' }
```

Actual output:
```
Output from task2 { payload: 'task1 payload' }
Output from task2 { payload: 'task2 payload' }
```

Another example, resulting in a disappearing handler:

```
let date = new Date();

date.setSeconds(date.getSeconds() + 3);

ManagerCron.registerTask('task1', date, { payload: 'task1 payload' })
    .then((task) => {
	    task.on('run', (data) => {
	    	console.log('Output from task1', data);
	    });
    });

date.setSeconds(date.getSeconds() + 3);

ManagerCron.registerTask('task2', date, { payload: 'task2 payload' })
	.then((task) => {
	    // Do not register a handler for this task
	});
```

Expected output:
```
Output from task1 { payload: 'task1 payload' }
```

Actual output:
```
(nothing)
```

**Comment by biemond** [comment]
yep , I see the same issue https://community.athom.com/t/2-device-crontasks-and-latest-added-one-is-run-twice-1st-is-ignored/9350

+1

**Comment by WeeJeWel** [comment]
Fixed in v2.0.5. Thank you for reporting.
